reana_job_controller/job_monitor.py

```diff
@@ -40,6 +39,33 @@ def __init__(self):
         self.job_event_reader_thread.daemon = True
         self.job_event_reader_thread.start()
 
+    def get_container_logs(self, last_spawned_pod):
+        """Get job pod's containers' logs."""
+        try:
+            job_log = \
+                current_k8s_corev1_api_client.read_namespaced_pod_log(
+                    namespace=last_spawned_pod.metadata.namespace,
+                    name=last_spawned_pod.metadata.name)
+            return job_log
+        except client.rest.ApiException as e:
+            if 'a container name must be specified for' in e.body:
```

Comment:
Looks good, I am only afraid of this line if the exception message content changes over time.

What do you think of doing, with the same amount of requests, something safer as the following?
- First request to get the Job Controller pod
- Then you iterate over all containers inside it and gather the logs

This way it will also gather the logs of additional sidecars if we create them in the future (this is kind of unlikely to happen but still), and also it wouldn't be tied to the fact that containers are called `krb5` and `job`, which we could change and silently break this feature (this I wouldn't say is likely to happen, but if it happens it would take potentially a lot of time to figure out if you don't know where to look).

Let me know what you think :)

Reply:
Completely agree ;)